main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1581,8 +1589,8 @@ object Weeder {
         case "==" => OperatorResult.BuiltIn(Name.mkQName("Eq.eq", sp1, sp2))
         case "!=" => OperatorResult.BuiltIn(Name.mkQName("Eq.neq", sp1, sp2))
         case "<=>" => OperatorResult.BuiltIn(Name.mkQName("Order.compare", sp1, sp2))
-        case "and" => OperatorResult.Operator(SemanticOp.BoolOp.And)
-        case "or" => OperatorResult.Operator(SemanticOp.BoolOp.Or)
+        case "and" => OperatorResult.BinaryOperator(SemanticOp.BoolOp.And)
+        case "or" => OperatorResult.BinaryOperator(SemanticOp.BoolOp.Or)
```

Comment:
I'm sorry to bother you with this, but I'm really confused. Can you please explain the difference between OperatorResult.BuiltIn and OperatorResult.BinaryOperator?

I also noticed something interesting while compiling in the Typer phase: OperatorResult.BuiltIn turns into TypedAst.Apply.

Reply:
Yes. Its important to understand that `+` is overloaded. This means it becomes a call to `Add.add` (take a look at `Add.flix` trait). Thus if you want to mutate such an operator, you have to look at the `Apply` node and make changes there.

Reply:
Thanks!